### DISCUSSION TITLE
Allow null-conditional access operator when accessing a pointer type

### DISCUSSION THREAD

**Discussion by john-h-k** [dotnet/csharplang#8822] [Language Ideas]
Currently, this code does not compile

```C#
class Foo
{
    public void* Bar;
}

Foo foo = ...;

void* p = foo?.Bar;
```

I suggest this should be legal, and return `null` if the left hand side is `null`, as is currently done with reference types

**Comment by Joe4evr** [comment]
Duplicate of #398.

**Comment by john-h-k** [comment]
That did not cover this operator explicitly 🙂 , it appeared they were suggesting `?.` on pointers as `?->`, rather than using `?.` where the resulting type is a pointer

**Comment by yaakov-h** [comment]
This does seem like an interesting case that isn't quite covered by #323 or #398.

To be clear, is the idea here that `notAPointerReference?.APointerReference` would become legal, and return a null pointer if `notAPointerReference` evalutes to null?

**Comment by john-h-k** [comment]
Yes, exactly @yaakov-h

**Comment by DanielShuey** [comment]
Any updates on this? Due to Unity I would be using ?? ??= ?. on pointers a lot, ref returns and ref structs are a pain. Thanks.

Oh yeah and allowing pointers to be used in Generic type parameters would be great too, thanks :).
